src/trace-agent.js

```diff
@@ -61,22 +61,6 @@ function TraceAgent(config, logger) {
 }
 
 /**
- * Halts this agent and unpatches any patched modules.
- */
-TraceAgent.prototype.stop = function() {
```

Comment:
What do you think about having this functionality in a method in `test/hooks/common.js` since stopping is really only needed now in testing?  Then test code could use `common.js` to tear down the agent where the agent was being stopped.  The benefit would be that then (with some changes to the e2e tests) I think we could get rid of the `forceNewAgent_` config option.